functions/src/create-user.ts

```diff
@@ -67,8 +68,9 @@ export const createUser = functions
       balance,
       totalDeposits: balance,
       createdTime: Date.now(),
-      totalPnLCached: 0,
-      creatorVolumeCached: 0,
+      profitCached: { daily: 0, weekly: 0, monthly: 0, allTime: 0 },
+      creatorVolumeCached: { daily: 0, weekly: 0, monthly: 0, allTime: 0 },
+      portfolioHistory: [] as PortfolioMetrics[],
```

Comment:
Love your solution! One thing I'd suggest though is to put the portfolio history into a user subcollection. e.g. `users/{userId}/portfolio`, with each doc containing the PortfolioMetrics.

This solves a lot of the scaling concerns because:
1. Firebase has a 1MB cap on the size of a doc.
2. User objects are public and downloaded by everyone frequently, so it's best not to store a lot in it anyway.
3. We can feel free to store as much data as we want, even on a 15 minute interval with little tradeoff.

Reply:
@jahooma Thanks for reviewing. Sounds good, I'll move that over to a subcollection. Great points!